rsocket-core/src/commonMain/kotlin/io/rsocket/Cancelable.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.rsocket
+
+import kotlinx.coroutines.*
+
+interface Cancelable {
+    val job: Job
+}
+
+val Cancelable.isActive: Boolean get() = job.isActive
+
+fun Cancelable.cancel(cause: CancellationException? = null) {
```

Comment:
curious: Why so many extension methods, instead of defaults on the interface.

Reply:
Mainly code style. But also, default methods can be overridden, so it's even safer to keep them as extensions.
Also, it's more [kotlinish](https://medium.com/@elizarov/extension-oriented-design-13f4f27deaee)

Reply:
I wouldn't say it's kotlin idiomatic, which is why I asked. But not an issue, just curious.